### PR TITLE
Outlook Display Samples

### PR DESCRIPTION
**Pull request by timwan10**
Adding the async versions of display functions. I decided that instead of creating whole new snippets, that it would be best to modify the existing one to have both examples of the display functions. As a result, I had to put the display directory under the preview/beta APIs. This can change back as soon as we ship 1.9.

### CODE REVIEW

File: config/build.ts [hunk 1]
```diff
@@ -267,7 +267,8 @@ async function processSnippets(processedSnippets: Dictionary<SnippetProcessedDat
 
         let snippetOfficeReferenceIsOk =
             officeJsReferences[0] === canonicalOfficeJsReference ||
-            (group.indexOf('preview-apis') >= 0 && officeJsReferences[0] === betaOfficeJsReference);
+            (group.indexOf('preview-apis') >= 0 && officeJsReferences[0] === betaOfficeJsReference) ||
+            (group.indexOf('display-items') >= 0 && officeJsReferences[0] === betaOfficeJsReference);
```
Comment (ElizabethSamuel-MSFT) [inline config/build.ts:271]: why this change?

Comment (ElizabethSamuel-MSFT) [reply config/build.ts:271]: Can you move the samples to the 99-preview-apis folder?

File: samples/outlook/55-display-items/display-reply-forms.yaml [hunk 2]
```diff
@@ -3,38 +3,59 @@ name: Create replies
...
 script:
     content: |
-        $("#displayReplyForm").click(displayReplyForm);
-        $("#displayReplyAllForm").click(displayReplyAllForm);
+        $("#display-reply-form").click(displayReplyForm);
+        $("#display-reply-form-async").click(displayReplyFormAsync);
+        $("#display-reply-all-form").click(displayReplyAllForm);
+        $("#display-reply-all-form-async").click(displayReplyAllFormAsync);
 
+        // Async functions are only currently available in preview.
```
Comment (ElizabethSamuel-MSFT) [inline samples/outlook/55-display-items/display-reply-forms.yaml:14]:
```suggestion
        // Async functions are only available in preview.
```

File: samples/outlook/55-display-items/display-new-message.yaml [hunk 3]
```diff
@@ -3,28 +3,50 @@ name: Create a new message
...
+                name: "image.png",
+                url: "http://www.cutestpaw.com/wp-content/uploads/2011/11/Cute-Black-Dogs-s.jpg",
+                isInline: true
+              }
+            ]
+          });
+        }
+
+        // The Async version is only in preview, and provides a callback when the new message form has been created
+        function runAsync() {
```
Comment (ElizabethSamuel-MSFT) [inline samples/outlook/55-display-items/display-new-message.yaml:30]:
```suggestion
        function runAsync() {
          // The async version is only available in preview, and provides a callback when the new message form has been created.
```

File: samples/outlook/55-display-items/display-existing-appointment.yaml [hunk 4]
```diff
@@ -3,18 +3,29 @@ name: Open an appointment
...
         function run() {
           var itemId = $("#itemId").val();
           Office.context.mailbox.displayAppointmentForm(itemId);
         }
+
+        // The Async version will return error(9049) if the item is not found
+        // The Async version is only in preview
+        function runAsync() {
+          var itemId = $("#itemId").val();
+          Office.context.mailbox.displayAppointmentFormAsync(itemId, function(asyncResult) {
```
Comment (ElizabethSamuel-MSFT) [inline samples/outlook/55-display-items/display-existing-appointment.yaml:25]: Can you make this change here and in other instances?
```suggestion
        function runAsync() {
          var itemId = $("#itemId").val();

          // The async version will return error 9049 if the item is not found.
          // The async version is only available in preview.
          Office.context.mailbox.displayAppointmentFormAsync(itemId, function(asyncResult) {
```

File: samples/outlook/55-display-items/display-new-appointment.yaml [hunk 5]
```diff
@@ -3,39 +3,66 @@ name: Create a new appointment
...
+            body: "Hello World!"
+          });
+        }
+
+        function runAsync() {
+          var start = new Date();
+          var end = new Date();
+          end.setHours(start.getHours() + 1);
+          
+          // The Async version is only in preview, and provides a callback when the new message form has been created
```
Comment (ElizabethSamuel-MSFT) [inline samples/outlook/55-display-items/display-new-appointment.yaml:34]:
```suggestion
          // The async version is only available in preview, and provides a callback when the new appointment form has been created.
```

File: samples/outlook/55-display-items/display-reply-with-attachments.yaml [hunk 6]
```diff
@@ -3,33 +3,45 @@ name: 'Create a reply, with attachments'
...
+            callback: function(result) {
+              if (result.status !== Office.AsyncResultStatus.Succeeded) {
+                console.error(`Action failed with message ${result.error.message}`);
+              }
+            }
+          });
+        }
+
+        function runAsync() {
+          Office.context.mailbox.item.displayReplyFormAsync(
```
Comment (ElizabethSamuel-MSFT) [inline samples/outlook/55-display-items/display-reply-with-attachments.yaml:29]: Can you add comment about this function being in preview?

File: config/build.ts (hunk 1 above)
Comment (timwan10) [reply config/build.ts]: Yeah, undid this change by moving to th e99-preview-apis folder.
